### PR TITLE
Retrieve first versioned object; needs coverage in object_store

### PR DESCRIPTION
**Pull request by patrickvinograd**

### CODE REVIEW

File: modules/vba_documents/spec/lib/object_store_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    @bucket = instance_double(Aws::S3::Bucket)
+    @object = instance_double(Aws::S3::Object)
+    allow(Aws::S3::Resource).to receive(:new).and_return(@resource)
+    allow(Aws::S3::Client).to receive(:new).and_return(@client)
+  end
+
+  describe '#bucket' do
+    it 'returns a bucket' do
+      expect(@resource).to receive(:bucket).and_return(@bucket)
+      described_class.new.bucket
```
Comment (jkassemi) [inline modules/vba_documents/spec/lib/object_store_spec.rb]: possibly worth checking the return value

File: modules/vba_documents/lib/vba_documents/object_store.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module VBADocuments
+  class ObjectStore
+    # Obtain the configured VBADocuments bucket
+    def bucket
```
Comment (jkassemi) [inline modules/vba_documents/lib/vba_documents/object_store.rb]: Is there a need for the `bucket` outside the instance?

File: modules/vba_documents/app/workers/vba_documents/upload_processor.rb [hunk 3]
```diff
@@ -72,9 +73,10 @@ def map_downstream_error(status, body)
     end
 
     def download_raw_file(guid)
```
Comment (jkassemi) [inline modules/vba_documents/app/workers/vba_documents/upload_processor.rb]: Might be worth a comment indicating that this returns a tempfile that should be closed. Would like to move toward providing documentation for vets-api internals via rdoc or something similar eventually.

Comment (jkassemi) [reply modules/vba_documents/app/workers/vba_documents/upload_processor.rb]: Probably worth confirming the caller currently closes the tempfile.

File: modules/vba_documents/lib/vba_documents/object_store.rb (hunk 2 above)
Comment (patrickvinograd) [reply modules/vba_documents/lib/vba_documents/object_store.rb]: Probably not, I'll clean that up along with refactoring a couple other classes to use the ObjectStore class

File: modules/vba_documents/app/workers/vba_documents/upload_processor.rb (hunk 3 above)
Comment (patrickvinograd) [reply modules/vba_documents/app/workers/vba_documents/upload_processor.rb]: Was not being explicit about it, fixed

**Pull request merged**